### ISSUE TITLE
ReactPDF.Image is not rendering some images

### ISSUE THREAD

**Issue by stepanve** [diegomura/react-pdf#2639]
**Describe the bug**
ReactPDF.Image is not rendering some images like the below image. there's no console error.
I confirmed that ReactPDF.Image is rendering other jpeg files or png. 
![304757573-06cae53d-85f0-4c9c-9066-f7dc0e819894](https://github.com/diegomura/react-pdf/assets/17928007/7ff00052-8e44-4d53-9153-100f7763ae4b)

**To Reproduce**
I've used ReactPDF like this.
```
  <ReactPDF.Image
    key={fileId}
    source={{
      uri: '/fae5c83b-431c-44e9-829d-004871a5f32f.jpeg',
      headers: { Pragma: 'no-cache', 'Cache-Control': 'no-cache' },
      method: 'GET',
      body: undefined,
    }}
    style={style.image}
  />
```

**Expected behavior**

**Desktop (please complete the following information):**

- OS: MacOS, W
- Browser chrome
- React-pdf version 3.3.8

**Comment by KMJ-007** [comment]
yes, i am also facing simillar issues

i tried downloading pdf from the playground

<img width="1440" alt="image" src="https://github.com/diegomura/react-pdf/assets/86996507/2f7506cd-d325-42f7-86aa-c9930f7f80cf">


but when i try to open using my mac(preview) or iPhone, i am getting blank image

<img width="958" alt="image" src="https://github.com/diegomura/react-pdf/assets/86996507/3c7958f0-559d-4729-a260-0d5946f4be8b">



Desktop (please complete the following information):

OS: MacOS 14.0
Browser Brave
React-pdf version 3.3.8

iPhone:
browser safari
React-pdf version 3.3.8


surprisingly second image from the playground renders perfectly

<img width="1090" alt="image" src="https://github.com/diegomura/react-pdf/assets/86996507/576cd414-8673-401a-95b8-f67e5ca0ce2e">

**Comment by mubin86** [comment]
I am also facing the exact same problem!! Do you have any ideas or any possible fix? @diegomura @wojtekmaj

**Comment by ooga** [comment]
@KMJ-007 your problem is fixed by #2646, but not sure if it's the same problem as the one originally mentioned by this issue

**Comment by mubin86** [comment]
@diegomura Still facing this weird issue. A few days ago actually I opened a issue similar to this one https://github.com/diegomura/react-pdf/issues/2651 and you actually closed my issue by providing a possible fix chore: bump jay-peg dep #2687 but the problem still persists with the latest version. Its happening for a long time so would you mind looking into it one more time please?

**Comment by stepanve** [comment]
Sorry for not keeping in touch.
I solved the problem when I updated @react-pdf/render from 3.1.9 to 3.4.2. Thank you for all committers.

Should I close this issue?

**Comment by hebontes** [comment]
Guys, as you can see we still have issues I tried updating to v3.4.2 but still does not work I tried using blob, base64 string, url nothing works, even something like this and it did not work help!!!!!

   <Image src={{ uri: image, method: "GET", headers: { "Cache-Control": "no-cache" }, body: "" }}/>

**Comment by mubin86** [comment]
I implemented a small workaround to resolve this issue, and it worked for me. Sharing it in case it might be helpful for others to try. FYI: in my case my problem was happening very rarely with a few of the JPEG images.

    let tempImage: HTMLImageElement;
    
    const resizeImage = (img: HTMLImageElement, maxWidth: number, maxHeight: number) => {
        const canvas = document.createElement('canvas');
        let width = img.width;
        let height = img.height;

        // https://stackoverflow.com/questions/2303690/resizing-an-image-in-an-html5-canvas
        // This is the reference resource from where actually got the concept of resizing image in HTML canvas

        /* istanbul ignore if */
        if (width > maxWidth) {
            height *= maxWidth / width;
            width = maxWidth;
        }
        /* istanbul ignore if */
        if (height > maxHeight) {
            width *= maxHeight / height;
            height = maxHeight;
        }

        canvas.width = width;
        canvas.height = height;
        const ctx = canvas.getContext('2d');
        ctx.drawImage(img, 0, 0, width, height);

        return canvas;
    };

    if (sampleData.picture) {
        tempImage = document.createElement('img');
        tempImage.setAttribute('data-testid', 'generate-pdf-picture');
        tempImage.crossOrigin = 'anonymous';

        tempImage.addEventListener('load', async () => {
            if (sampleData.picture.mimeType && sampleData.picture.mimeType !== 'image/png') {
                const maxWidth = 1000;
                const maxHeight = 800;
                const resizedCanvas = resizeImage(tempImage, maxWidth, maxHeight);

                const pngDataUrl = resizedCanvas.toDataURL('image/png', 0.8);

                // To mitigate the unexpected library issue and prevent disappearance of the image upon download, convert it to PNG format
                sampleData.picture.imageSrc = pngDataUrl;
            }
        });

        tempImage.src = sampleData.picture.imageSrc;
    }

**Comment by nathanlao** [comment]
Currently I am using "@react-pdf/renderer": "^3.4.4", seems like the problem still exists.

**Comment by Friendseeker** [comment]
Hey guys if you have other problematic images that cannot be displayed, would you guys like to share these images?

I also got hit with the same issue and found out the root cause is with https://github.com/diegomura/jay-peg not able to parse the image EXIF metadata properly. I made a [PR](https://github.com/diegomura/jay-peg/pull/11) to address some of the issues I identified, however with more images me or other volunteers might be able to fix other outstanding issues with jay-peg.

**Comment by RakeshSandila** [comment]
try with png format images they are working fine

**Comment by Lamba15** [comment]
Hey guys, so I had the same Issue and have been working on it for days until I discovered the solution by acedient while trying to test why some images work and some others don't work.

basically images must be bigger than the size they are rendered at, the exact ratio I didn't calculate, but using big images or rendering small images with even even smaller res for example the image wxh 1000x1000 I make the image tag 500x500

TLDR;
increase image W*H than how you render them it will work fine.

**Comment by mbhartleyza** [comment]
Hi.   I found that in my case.  The PNG wouldn't show when it was less than 100kb.   I used Photopea to make the image a little bigger and it now shows up.  
I was also getting the error message, file format not supported.  Which was strange because I had other PNG files that worked.

**Comment by BINARY-TECH001** [comment]
I'm also working with react-pdf but no matter what i tried, images won't show - i've prompt chatgpt and all its saying is to convert to base64 which yiled no result. 

I've even tried to cache the image cuz i thought the reatc pdf <Image /> component would try make a fetch which supposed to be fine but still nothing - The only thing that works is local image which is having images locally in my public nextjs public folder but the images i wanted to use are dynamic coming from api

**Comment by lilliekittredge** [comment]
This is still occurring.   I can't share the test image that is failing for me because it belongs to my customer but it is a PNG,  594 × 478, 30,857 bytes.   Other images work fine.   My source has the data inline, it's not a file coming from elsewhere. 

Loving the product btw, once I found you this project went way faster than I expected!

**Comment by lilliekittredge** [comment]
Turns out in my case the pngs in question were corrupt.  They display fine in the web but I noticed they don't have a thumbnail in the OS.  Converting to jpg and back fixed it for me.
